sentry/src/main/java/io/sentry/transport/RateLimiter.java

```diff
@@ -21,42 +23,25 @@
 /** Controls retry limits on different category types sent to Sentry. */
 public final class RateLimiter {
 
-  private enum DataCategory {
-    All("__all__"),
-    Default("default"), // same as Error
-    Error("error"),
-    Session("session"),
-    Attachment("attachment"),
-    Transaction("transaction"),
-    Security("security"),
-    Unknown("unknown");
-
-    private final String category;
-
-    DataCategory(final @NotNull String category) {
-      this.category = category;
-    }
-
-    public String getCategory() {
-      return category;
-    }
-  }
-
   private static final int HTTP_RETRY_AFTER_DEFAULT_DELAY_MILLIS = 60000;
 
   private final @NotNull ICurrentDateProvider currentDateProvider;
   private final @NotNull ILogger logger;
+  private final @NotNull SentryOptions options;
   private final @NotNull Map<DataCategory, @NotNull Date> sentryRetryAfterLimit =
       new ConcurrentHashMap<>();
 
   public RateLimiter(
-      final @NotNull ICurrentDateProvider currentDateProvider, final @NotNull ILogger logger) {
+      final @NotNull ICurrentDateProvider currentDateProvider,
+      final @NotNull ILogger logger,
+      final @NotNull SentryOptions options) {
     this.currentDateProvider = currentDateProvider;
     this.logger = logger;
+    this.options = options;
   }
 
-  public RateLimiter(@NotNull ILogger logger) {
-    this(CurrentDateProvider.getInstance(), logger);
+  public RateLimiter(final @NotNull ILogger logger, final @NotNull SentryOptions options) {
```

Comment:
Nit: Since we are injecting options now, we could access the logger from it and remove it as a ctor param and instance member.

Reply:
removed `logger` param and field